### PR TITLE
Form type content type

### PR DESCRIPTION
**Pull request by itkg-alavieille**

### CODE REVIEW

File: BackofficeBundle/Form/DataTransformer/ChoicesOptionToArrayTransformer.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     * @param SuppressSpecialCharacterTransformer $suppressSpecialCharacterClass
+     */
+    public function __construct(SuppressSpecialCharacterTransformer $suppressSpecialCharacterClass)
+    {
+        $this->suppressSpecialCharacterClass = $suppressSpecialCharacterClass;
+    }
+
+    /**
+     * @param array $arrayChoices
+     * @return string
```
Comment (nicolasThal) [inline BackofficeBundle/Form/DataTransformer/ChoicesOptionToArrayTransformer.php]: empty line before the @return tag

File: BackofficeBundle/Form/DataTransformer/ChoicesOptionToArrayTransformer.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        if (null === $arrayChoices) {
+            return "";
+        }
+
+        return implode(',', $arrayChoices);
+    }
+
+    /**
+     * @param string $choices
+     * @return array choix1,choix2,choix3
```
Comment (nicolasThal) [inline BackofficeBundle/Form/DataTransformer/ChoicesOptionToArrayTransformer.php]: just put the @return array tag

If you want to be more clear, either the test should do it or you can write a little comment about it

File: BackofficeBundle/Tests/Form/DataTransformer/ChoicesOptionToArrayTransformerTest.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+{
+    protected $suppressSpecialCharacter;
+    protected $transformer;
+
+    /**
+     * Set up the test
+     */
+    public function setUp()
+    {
+        $this->suppressSpecialCharacter = new SuppressSpecialCharacterTransformer();
```
Comment (nicolasThal) [inline BackofficeBundle/Tests/Form/DataTransformer/ChoicesOptionToArrayTransformerTest.php]: your test is not isolated, you should Phake the `SuppressSpecialCharacterTransformer` and write a test (if not already done) for it beside

File: BackofficeBundle/Resources/config/transformer.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ parameters:
     open_orchestra_backoffice.transformer.embedkeywords_to_keywords.class: OpenOrchestra\BackofficeBundle\Form\DataTransformer\EmbedKeywordsToKeywordsTransformer
     open_orchestra_backoffice.transformer.video_url_to_id.class: OpenOrchestra\BackofficeBundle\Form\DataTransformer\VideoUrlToIdTransformer
     open_orchestra_backoffice.transformer.suppress_special_character.class: OpenOrchestra\BackofficeBundle\Form\DataTransformer\SuppressSpecialCharacterTransformer
-
+    open_orchestra_backoffice.transformer.choices_option_to_array.class: OpenOrchestra\BackofficeBundle\Form\DataTransformer\ChoicesOptionToArrayTransformer
 services:
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/config/transformer.yml]: a newline between parameters and service definitions is welcomed :)

File: BackofficeBundle/Resources/translations/messages.fr.yml [hunk 5]
```diff
@@ -202,6 +206,17 @@ open_orchestra_backoffice:
...
+            multiple: Multiple
+            expanded: Etendu
+            currency: Devise
+            precision: Précision
+            date_format: Format de la date
+            widget: Widget
+            widget_type:
+                choice: Listes déroulantes
+                text: Plusieurs champs texte
+                single_text: Un Champ texte
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/translations/messages.fr.yml]: pas de majuscule à champ pour être cohérent avec le libellé précédent

**Comment by itkg-alavieille** [comment]
@itkg-ngilain @nicolasThal

**Comment by itkg-ngilain** [comment]
:+1: don't forget to rebase before merging

File: BackofficeBundle/Form/Type/OrchestraChoicesOption.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ * Class OrchestraChoicesOption
+ */
+class OrchestraChoicesOption extends AbstractType
+{
+    protected $choiceTransformer;
+
+    /**
+     * @param ChoicesOptionToArrayTransformer $choiceTransformer
+     */
+    public function __construct(ChoicesOptionToArrayTransformer $choiceTransformer){
```
Comment (nicolasThal) [inline BackofficeBundle/Form/Type/OrchestraChoicesOption.php]: PSR, the { should be on a new line

File: BackofficeBundle/Form/Type/OrchestraChoicesOption.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        $this->choiceTransformer = $choiceTransformer;
+    }
+
+    /**
+     * @param FormBuilderInterface $builder
+     * @param array $options
```
Comment (nicolasThal) [inline BackofficeBundle/Form/Type/OrchestraChoicesOption.php]: can you alin your parameters

File: BackofficeBundle/Form/Type/OrchestraChoicesOption.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->addModelTransformer($this->choiceTransformer);
+    }
+
+    public function getParent()
```
Comment (nicolasThal) [inline BackofficeBundle/Form/Type/OrchestraChoicesOption.php]: you forgot the class comment

File: BackofficeBundle/Tests/Form/DataTransformer/ChoicesOptionToArrayTransformerTest.php (hunk 3 above)
Comment (nicolasThal) [inline BackofficeBundle/Tests/Form/DataTransformer/ChoicesOptionToArrayTransformerTest.php]: your test is still not isolated, you should Phakee this class

Comment (nicolasThal) [inline BackofficeBundle/Tests/Form/DataTransformer/ChoicesOptionToArrayTransformerTest.php]: this line should be suppressed as you use a mock

**Comment by itkg-alavieille** [comment]
@nicolasThal

File: BackofficeBundle/Tests/Form/DataTransformer/ChoicesOptionToArrayTransformerTest.php (hunk 3 above)
Comment (nicolasThal) [inline BackofficeBundle/Tests/Form/DataTransformer/ChoicesOptionToArrayTransformerTest.php]: still this line is to suppress!!

**Comment by nicolasThal** [comment]
:+1: once the line is deleted!!!

**Pull request merged**